packages/xsnap/test/test-xsnap.js

```diff
@@ -511,3 +511,28 @@ test('property name space exhaustion: orderly fail-stop', async t => {
     });
   }
 })();
+
+test('round-trip byte sequences from golang for dynamic IBC', async t => {
+  const opts = options();
+  const vat = xsnap(opts);
+  t.teardown(() => vat.terminate());
+
+  await vat.evaluate(`
+    globalThis.send = it => issueCommand(ArrayBuffer.fromString(JSON.stringify(it)));
+  `);
+  function checkStrings() {
+    const octets = new Array(256).fill(0).map((_, i) => i);
+    // We receive some external JSON data (from Golang) in the following format:
+    const bstring1 = String.fromCharCode.apply(null, octets);
+    const bstring2 = JSON.parse(JSON.stringify(bstring1));
```

Comment:
I thought of a further test that may be essential to the situation:

```suggestion
        const jstring1 = JSON.stringify(bstring1);
        const joctets = new Array(256).fill(0).map((_, i) => `\\x${i.toString(16).leftPad(2, '0')}`).join('')
        assert(jstring1 === `"${joctets}"`);
        const bstring2 = JSON.parse(jstring2);
```

Please forgive the untested well-intentions here.  I'm at the end of my day and don't have a clearer way to generate `"\\x00\\x01\\x02...\\xfe\\xff"`.  Perhaps it should be written out in full, for clarity of the test.